### PR TITLE
Fixes #1205 - Able to schedule for past times in current date

### LINKED ISSUE #1205: Able to schedule for past times in current date
### Issue Summary
Past times for the current date are available for booking.

### Steps to Reproduce

1. Create an event type
2. Go to booking website
3. Select current date

### Technical details

- Browser version: Chrome 96

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

Fixes #1205

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Open AvailabilityTimes on the booking page, play around with your availability settings.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/slots.ts [hunk 1]
```diff
@@ -17,37 +19,41 @@ export type GetSlots = {
 
 const getMinuteOffset = (date: Dayjs, step: number) => {
   // Diffs the current time with the given date and iff same day; (handled by 1440) - return difference; otherwise 0
-  const minuteOffset = Math.min(date.diff(dayjs().startOf("day"), "minutes"), 1440) % 1440;
+  const minuteOffset = Math.min(date.diff(dayjs().startOf("day"), "minute"), 1440) % 1440;
```
Comment (emrysal) [inline lib/slots.ts:22]: Singular as this is documented, plural is undocumented; you never know when there is a breaking change to remove support for pluralization.

File: lib/slots.ts [hunk 2]
```diff
@@ -17,37 +19,41 @@ export type GetSlots = {
...
   // round down to nearest step
-  return Math.floor(minuteOffset / step) * step;
+  return Math.ceil(minuteOffset / step) * step;
```
Comment (emrysal) [inline lib/slots.ts:24]: If the time was 13:37 flooring caused the last past slot to be available, ceil to resolve.

File: lib/slots.ts [hunk 3]
```diff
@@ -17,37 +19,41 @@ export type GetSlots = {
...
 const getSlots = ({ inviteeDate, frequency, minimumBookingNotice, workingHours }: GetSlots) => {
   // current date in invitee tz
-  const startDate = dayjs(inviteeDate).add(minimumBookingNotice, "minutes"); // + minimum notice period
+  let startDate = dayjs(inviteeDate).add(minimumBookingNotice, "minute");
   // checks if the start date is in the past
   if (startDate.isBefore(dayjs(), "day")) {
     return [];
   }
+  // Add the current time to the startDate if the day is today
+  if (startDate.isToday()) {
```
Comment (emrysal) [inline lib/slots.ts:35]: All inviteeDates are exactly that; dates. They don't know about the current time. But if the date is today we do want to take the current time into account.

File: lib/slots.ts [hunk 4]
```diff
@@ -17,37 +19,41 @@ export type GetSlots = {
...
-      endTime: dayjs.utc().startOf("day").add(schedule.endTime, "minutes"),
+      startTime: dayjs.utc().startOf("day").add(schedule.startTime, "minute"),
+      endTime: dayjs.utc().startOf("day").add(schedule.endTime, "minute"),
     }))
   ).filter((hours) => hours.days.includes(inviteeDate.day()));
 
   const slots: Dayjs[] = [];
-  for (let minutes = getMinuteOffset(inviteeDate, frequency); minutes < 1440; minutes += frequency) {
-    const slot = inviteeDate.startOf("day").add(minutes, "minutes");
+  for (let minutes = getMinuteOffset(startDate, frequency); minutes < 1440; minutes += frequency) {
```
Comment (emrysal) [inline lib/slots.ts:49]: Never use inviteeDate - startDate takes booking notice into account whereas inviteeDate does not.

**Review by pumfleet — APPROVED** [review]
Thanks for this fix. Your comments were super helpful

**Pull request merged**